app/controllers/users_controller.rb

```diff
@@ -1,3 +1,28 @@
 class UsersController < ApplicationController
-  def new; end
+  def show
+    @user = User.find_by id: params[:id]
+    return if @user
+    flash[:danger] = t "users.user_not_found"
+    redirect_to signup_path
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new user_params
+    if @user.save
+      flash[:success] = t "static_pages.welcome"
+      redirect_to @user
+    else
+      render :new
+    end
+  end
+
+  private
+
+    def user_params
```

Comment:
Please remove 2 space indentation from the left, because after the private keyword you do not need to add 2 space indentation on the left.